packages/lit-html/src/lib/lit-html.ts

```diff
@@ -12,9 +12,557 @@
+    // regexes
+    let regex = textRegex;
+
+    for (let i = 0; i < strings.length - 1; i++) {
+      const s = strings[i];
+      // The index of the end of the last attribute. When this is !== -1 at
+      // end of a string, it means we're in q quoted attribute position.
+      let attrNameEnd = -1;
+      let lastIndex = 0;
+
+      // The conditions in this loop handle the current parse state, and the
+      // assignments to the `regex` variable are the state transitions.
+      while (lastIndex < s.length) {
+        // Make sure we start searching from where we previously left off
+        regex.lastIndex = lastIndex;
+        const match = regex.exec(s);
+        if (match === null) {
+          // TODO (justfagnani): add test coverage from spread parts
+          // If we're not in a quoted attribute value, make sure we clear
+          // the attrNameEnd marker
+          if (regex !== singleQuoteAttr && regex !== doubleQuoteAttr) {
+            attrNameEnd = -1;
+          }
+          break;
+        }
+        lastIndex = regex.lastIndex;
+        if (regex === textRegex) {
+          // TODO (justinfagnani): there are more ways to open a comment in
+          // HTML, like `<?` or `</{`. Figure out whether / how to handle that.
+          if (match[1] === '!--') {
+            regex = commentRegex;
+          } else {
+            if (rawTextElement.test(match[1])) {
+              regex = new RegExp(`<\/${match[1]}`);
+            } else {
+              regex = tagRegex;
+            }
+          }
+        } else if (regex === commentRegex) {
+          regex = textRegex;
+        } else if (
+          regex === tagRegex ||
+          regex === doubleQuoteAttr ||
+          regex === singleQuoteAttr
+        ) {
+          if (match[1] === '>') {
+            regex = textRegex;
+          } else if (quote && (match[0] === '"' || match[0] === "'")) {
+            // End quoted attribute
+            attrNameEnd = -1;
+            regex = tagRegex;
+          } else {
+            attrNameEnd = regex.lastIndex - match[3].length;
+            quote = match[5];
+            regex =
+              quote === undefined
+                ? tagRegex
+                : quote === '"'
+                ? doubleQuoteAttr
+                : singleQuoteAttr;
+          }
+        } else {
+          // Not one of the five state regexes, so it must be the dynamically
+          // created raw text regex and we're at the close of that element.
+          regex = tagRegex;
+        }
+      }
+
+      html +=
+        regex === textRegex
+          ? s + nodeMarker
+          : (attrNameEnd === -1
+              ? s
+              : s.slice(0, attrNameEnd) + '$lit$' + s.slice(attrNameEnd)) +
+            marker;
+    }
+
+    // TODO (justinfagnani): if regex is not textRegex log a warning for a
+    // malformed template in dev mode.
+
+    this.__element.innerHTML =
+      html + this.__strings[l] + (type === 2 ? '</svg>' : '');
```

Comment:
If we wanna get sloppy, we don't actually need to close the `<svg>`.

Reply:
I like it!